src/transformers/generation/configuration_utils.py

```diff
@@ -44,6 +53,16 @@
 METADATA_FIELDS = ("_from_model_config", "_commit_hash", "_original_object_hash", "transformers_version")
 NEEDS_CACHE_CONFIG = {}
 
+NEED_SETUP_CACHE_CLASSES_MAPPING = {
```

Comment:
moved these constants to the file holding `GenerationConfig`, so we can validate cache config options